File: frontend/admin/src/types/location.ts
import { PaginationFilter } from "./common";

export type Location = {
  id: number;
  code?: string | null;
  name: string;
  email?: string | null;
  phone?: string | null;
  country?: string;
  country_code?: string | null;
  province?: string;
  province_code?: string | null;
  district?: string;
  district_code?: string | null;
  ward?: string;
  ward_code?: string | null;
  address1?: string | null;
  address2?: string;
  zip?: string;
  fulfill_order: boolean;
  inventory_management?: boolean;
  deactive_inventory_at?: Date;
  default_location: boolean;
  offline_store: boolean;
  start_date?: Date | null;
  end_date?: Date | null;
  created_on?: Date;
  modified_on?: Date;
  status: LocationStatus;
};

export type LocationPriority = Location & {
  rank: number;
};

export type LocationFilter = PaginationFilter & {
  query?: string;
  ids?: string;
  inventory_management?: boolean | null;
  fulfill_order?: boolean | null;
  statuses?: LocationStatus[];
};

export type LocationStatus = "active" | "expired";

File: backend/src/main/java/vn/hust/omni/sale/service/store/domain/model/Location.java
package vn.hust.omni.sale.service.store.domain.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "Locations")
@DynamicUpdate
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Location {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private int storeId;
    private String code;
    private String name;
    private String email;
    private String phone;
    private boolean defaultLocation;

    private String country;
    private String countryCode;
    private String province;
    private String provinceCode;
    private String district;
    private String districtCode;
    private String ward;
    private String wardCode;
    private String address1;

    private boolean deleted;
    private boolean inventoryManagement;
    @Enumerated(EnumType.STRING)
    private Status status;

    @CreationTimestamp
    private Instant createdOn;
    @UpdateTimestamp
    private Instant modifiedOn;

    public Location(int id, int storeId, String code) {
        this.id = id;
        this.code = code;
        this.storeId = storeId;
    }

    public enum Status {
        active, expired
    }
}
